### ISSUE TITLE
Figure out how exceptions are used in MPSGE

### ISSUE THREAD

**Issue by davidanthoff** [anthofflab/MPSGE.jl#38]

**Comment by EliLazarus** [comment]
We should add error messages to catch user errors like not allocating taxes to an agent, or having variables that are never used.

**Comment by EliLazarus** [comment]
Figure out how exceptions are used in MPSGE (and implement those).
We should check that each input/enowment is also an output. (I was thinking we could check that each commodity is in balance in the benchmark, but maybe that's a lot of work replicating the over all benchmark balance?

**Comment by mitchphillipson** [comment]
This is the source/sink issue. This _might_ be a non-issue now, but it's something we should test.

Anecdotally, I've been stopped by MPSGE when my model was bad. This is a good thing.

**Comment by EliLazarus** [comment]
Yes, I think some good elements to match here, sink/source for sure (with clear error messages).

And maybe a separate issue, and maybe it's just a documentation question ("if your model isn't balanced, run generate_report and sort by margin"), but potentially a method to use the margin values to directly point to what elements of a model aren't balanced.

**Comment by mitchphillipson** [comment]
As a note, it's not just margin. It's margin*variable, that's what you want
to be zero. It just happens that most of the time variable != 0.

Also fixed variables may be different, but I need to test more.
